compiler/frontend/src/org/jetbrains/kotlin/psi/KtCodeFragment.kt

```diff
@@ -67,7 +73,15 @@ abstract class KtCodeFragment(
 
     override fun isValid() = true
 
-    override fun getContext() = context
+    override fun getContext(): PsiElement? {
+        if (fakeContextForJavaFile != null) return fakeContextForJavaFile
+        if (context !is KtElement) {
+            LOG.warn("CodeFragment with non-kotlin context should have fakeContextForJavaFile set: \noriginalContext = ${context?.getElementTextWithContext()}")
```

Comment:
We shouldn't create kotlin codeFragments for scala files. See KotlinCodeFragmentFactory.isContextAccepted